src/test/java/com/gildedrose/GildedRoseTest.java

```diff
@@ -2,16 +2,160 @@
 
 import static org.junit.Assert.*;
 
+import org.junit.Before;
 import org.junit.Test;
 
 public class GildedRoseTest {
 
+    private static final String ITEM_DEFAULT_NAME = "foo";
+    private static final int ITEM_DEFAULT_SELLIN = 15;
+    private static final int ITEM_DEFAULT_QUALITY = 15;
+    private static final int ITEM_OVERDUE_SELLIN = 0;
+    private static final int ITEM_LONG_OVERDUE_SELLIN = -50;
+    private static final int ITEM_QUALITY_MAX = 50;
+    private static final String ITEM_NAME_BACKSTAGE_PASS = "Backstage passes to a TAFKAL80ETC concert";
+    private static final String ITEM_NAME_AGED_BRIE = "Aged Brie";
+    private static final String ITEM_NAME_SULFURAS = "Sulfuras, Hand of Ragnaros";
+    private static final int ITEM_QUALITY_MIN = 0;
+    private Item item;
+    private GildedRose app;
+
+    @Before
+    public void setUp() {
+        item = new Item(ITEM_DEFAULT_NAME, ITEM_DEFAULT_SELLIN, ITEM_DEFAULT_QUALITY);
+        final Item[] items = new Item[] {item};
+        app = new GildedRose(items);
+    }
+
+    @Test
+    public void doesNotAlterName() {
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_NAME, app.items[0].name);
+    }
+
+    @Test
+    public void doesDecreaseSellIn() {
+        app.updateQuality();
+        assertEquals(ITEM_DEFAULT_SELLIN - 1, app.items[0].sellIn);
```

Comment:
yeah, we were talking about that. we could either use `item` all the time, or `app.items[0]` all the time, as they reference the very same object based on the current implementation of `GuildedRose`.

i personally would have preferred something like this (no setup, or global constants and hidden values, but a single helper method):
```java
@Test public void myTest() {
  final Item item = new Item(GENERICA, NEGATIVE_SELL_IN, ANY_QUALITY);

  update(item); // hides execution of SUT which got side effects and operates on mutable Item object

  assertEquals(NEGATIVE_SELL_IN - 1, item.sellIn);
}
```